### ISSUE TITLE
[AC] [Feature] More NPCBot Hiring Services

### ISSUE THREAD

**Issue by Conan1996-2** [trickerer/Trinity-Bots#541]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that my issue may get **closed without notice** if I *intentionally* remove or skip any mandatory\* field
- [X] This is NOT a bug/crash report
- [X] I've searched the bugtracker for similar questions/requests **including closed ones**

### Description

Can I create a 8 more NPCBots hiring services (70000), one for each major city. If not, can something like this be implemented.

### TrinityCore or AzerothCore

Both

### Core rev. hash/commit

AzerothCore rev. 43f46dc5eb0c+ 2023-06-24 23:31:18 +0000 (master branch) (Unix, RelWithDebInfo, Static) (worldserver-daemon)

### Operating system

Ubuntu 22.04

**Comment by Russianranger** [comment]
You can, Dinkledork aka @Day36512 has done this very thing in his repack. He had multiple npcbot hiring NPCs, "Fink" which allow you to hire NPC bots at every capital and starting area, and probably other areas as well. You'll have to ask him how exactly he did it, but it's probably something easy to replicate.

**Comment by trickerer** [comment]
Yes, you can - that is its purpose actually. 70000 is a normal NPC which can be spawned anywhere.

**Comment by Conan1996-2** [comment]
Perfect, and works great for Alliance... but Horde on the other hand have no hiring services... the Bot when spawned only has alliance characters.

**Comment by trickerer** [comment]
Botgiver only offers bots already spawned in world.

**Comment by Day36512** [comment]
@Conan1996-2 

Can't upload sql files in comments directly but you can run these queries in your 'creature' table. Will spawn the hire bot in every major city and other cities throughout the world:

[HireAllOver.txt](https://github.com/trickerer/Trinity-Bots/files/11875339/HireAllOver.txt)

**Comment by Conan1996-2** [comment]
Day36512, Perfect Thank you.

Trickerer, can the Hire bots be filtered for the faction your playing, so Alliance are not running with Horde bots and vice versa?

**Comment by trickerer** [comment]
> Trickerer, can the Hire bots be filtered for the faction your playing, so Alliance are not running with Horde bots and vice versa?

Look for it in config. `NpcBot.Bitgiver.FilterRaces`

**Comment by PREngineer** [comment]
> @Conan1996-2
> 
> Can't upload sql files in comments directly but you can run these queries in your 'creature' table. Will spawn the hire bot in every major city and other cities throughout the world:
> 
> [HireAllOver.txt](https://github.com/trickerer/Trinity-Bots/files/11875339/HireAllOver.txt)

@Day36512

Apologies for bringing this up.  I was searching a way to do this and found this old issue.  The table seems to be different now.  Is it OK if I just remove the extra IDs that don't exist in the table?  What are those supposed to be?  I am using the "Trinity with Bots" version of TC instead of the repack.

![1](https://github.com/user-attachments/assets/a93298eb-157b-48a0-a0fc-db5f1c847442)

Many thanks.

**Comment by Day36512** [comment]
My share was intended for azerothcore.

**Comment by PREngineer** [comment]
> My share was intended for azerothcore.

Hi sir!  Thanks for your quick reply.  I wasn't aware that the databases were so different between cores.

It does work in Trinity Core 3.3.5.

I checked and the `guid`s were not in use.  I removed the extra 2 `id` columns and executed against my TC 3.3.5a.  Now, the characters are showing up nicely.

![1](https://github.com/user-attachments/assets/7c57ee40-ae21-4703-a857-45f9f0209756)

Here's my file with the queries modified.

[Bot Representatives.txt](https://github.com/user-attachments/files/16324854/Bot.Representatives.txt)

Thanks for sharing!
